core/src/main/scala/kafka/cluster/Broker.scala

```diff
@@ -93,41 +106,53 @@ object Broker {
    *               id (int), number of endpoints (int), serialized endpoints
    * @return broker object
    */
-  def readFrom(buffer: ByteBuffer): Broker = {
+  def readFrom(buffer: ByteBuffer, expectRack: Boolean = true): Broker = {
     val id = buffer.getInt
     val numEndpoints = buffer.getInt
 
     val endpoints = List.range(0, numEndpoints).map(i => EndPoint.readFrom(buffer))
             .map(ep => ep.protocolType -> ep).toMap
-    new Broker(id, endpoints)
+    if (expectRack) {
+      val rack = readShortString(buffer)
+      new Broker(id, endpoints, Option(rack))
+    } else {
+      new Broker(id, endpoints, Option.empty)
+    }
   }
 }
 
-case class Broker(id: Int, endPoints: Map[SecurityProtocol, EndPoint]) {
+case class Broker(id: Int, endPoints: Map[SecurityProtocol, EndPoint], rack: Option[String]) {
+
+  override def toString: String = id + " : " + endPoints.values.mkString("(",",",")") + ":" + rack.orNull
 
-  override def toString: String = id + " : " + endPoints.values.mkString("(",",",")")
+  def this(id: Int, endPoints: Map[SecurityProtocol, EndPoint]) = {
+    this(id, endPoints, Option.empty)
+  }
 
   def this(id: Int, host: String, port: Int, protocol: SecurityProtocol = SecurityProtocol.PLAINTEXT) = {
-    this(id, Map(protocol -> EndPoint(host, port, protocol)))
+    this(id, Map(protocol -> EndPoint(host, port, protocol)), Option.empty)
   }
 
   def this(bep: BrokerEndPoint, protocol: SecurityProtocol) = {
     this(bep.id, bep.host, bep.port, protocol)
   }
 
-
-  def writeTo(buffer: ByteBuffer) {
+  def writeTo(buffer: ByteBuffer, includeRack: Boolean = true) {
     buffer.putInt(id)
     buffer.putInt(endPoints.size)
     for(endpoint <- endPoints.values) {
       endpoint.writeTo(buffer)
     }
+    if (includeRack) {
+      writeShortString(buffer, rack.orNull)
+    }
   }
 
-  def sizeInBytes: Int =
+  def sizeInBytes(includeRack: Boolean = true): Int =
```

Comment:
Broker.sizeInBytes() and Broker.writeTo() was used for serialization and deserialization of UpdateMetdataRequest when I started the PR. That's why I have to add includeRack parameter for version compatibility.  It was recently changed to use the Java code in kafka.common for this. But Broker.readFrom and Broker.writeTo remain in the code. So I am not sure if they are still needed. If not, we can delete this part of code all together.